framework/wazuh/rule.py

```diff
@@ -206,7 +208,7 @@ def get_rules_files(status: str = None, relative_dirname: str = None, filename:
 
     data = process_array(rules_files, search_text=search_text, search_in_fields=search_in_fields,
                          complementary_search=complementary_search, sort_by=sort_by, sort_ascending=sort_ascending,
-                         offset=offset, limit=limit)
+                         offset=offset, limit=limit, q=q)
```

Comment:
When using `q` with any field different than `name`, I'm getting an error. Could you check this out?

> GET /groups?q=mergedSum%3Ddefault

```JSON
{
  "title": "Bad Request",
  "detail": "Field does not exist.: Available fields: name. Field: mergedSum",
  "dapi_errors": {
    "master-node": {
      "error": "Field does not exist.: Available fields: name. Field: mergedSum"
    }
  },
  "error": 1408
}
```

These are common returned fields:
```JSON
{
  "data": {
    "affected_items": [
      {
        "name": "default",
        "count": 2,
        "mergedSum": "4a8724b20dee0124ff9656783c490c4e",
        "configSum": "ab73af41699f13fdd81903b5f23d8d00"
      }
    ],
    "total_affected_items": 1,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected groups information was returned",
  "error": 0
}
```

If the fix is not that simple, open a new issue to improve the `process_array` function.

Reply:
This is expected. The `group` table has only one field (`name`) so that's the only one we can use to query rows:

https://github.com/wazuh/wazuh/blob/a7b092565dbd3f48e64bcfa77d8d2087431b0fb0/framework/wazuh/core/agent.py#L273

The `mergedSum` field is not in the table and is added after the query:

https://github.com/wazuh/wazuh/blob/a7b092565dbd3f48e64bcfa77d8d2087431b0fb0/framework/wazuh/agent.py#L591-L607

This is the  exact same problem I faced with the `group` and `group_hash` [issue](https://github.com/wazuh/wazuh/issues/16285). The filtering, sorting and selection is tied up to the fields that the table of the object has and other fields are rejected.

Reply:
As we agreed after discussing, we can't add these filters to the endpoint if they won't work. I'm moving this to `On hold` until it's also discussed with the Frontend team and something is decided.